eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/ExplorerApi.scala

```diff
@@ -163,7 +166,7 @@ object ExplorerApi {
             val JInt(nonce) = block \ "nonce"
             val previousBlockHash = (block \ "prev_block").extractOpt[String].map(ByteVector32.fromValidHex(_).reverse).getOrElse(ByteVector32.Zeroes)
             val merkleRoot = (block \ "mrkl_root").extractOpt[String].map(ByteVector32.fromValidHex(_).reverse).getOrElse(ByteVector32.Zeroes)
-            val header = BlockHeader(version.toLong, previousBlockHash, merkleRoot, OffsetDateTime.parse(time).toEpochSecond, bits.toLong, nonce.toLong)
+            val header = new BlockHeader(version.toLong, previousBlockHash, merkleRoot, OffsetDateTime.parse(time).toEpochSecond, bits.toLong, nonce.toLong)
```

Comment:
I think we should have a scala `BlockHeader` type in `bitcoin-lib` to hide the kotlin part.

Reply:
There are 2 possible design choices here:

1. make the transition from bitcoin-lib as easy as possible: add many wrappers, conversion methods, ...
2. make the transition to bitcoin-kmp as easy as possible and start using bitcoin-kmp types as much as possible (and update bitcoin-kmp if needed, being easily usable from Scala and Java is one of its primary design goals)

I'm leaning towards 2, which is why there are no scala wrappers for blocks, headers, psbt, base58, bech32...
There is an exception for transactions and scripts because otherwise migration would have been too difficult and clumsy. 

But adding wrappers for blocks and headers is easy and would have no negative impacts that I can see (memory, performance...) so I'll wait a bit and see what the general feeling about this is.

Reply:
I think you're right, I'm starting to lean a bit more towards option 2, I guess I just need to spend more time playing around in the codebase and see how it impacts the dev experience.